skxray/fitting/xrf_model.py

```diff
@@ -1085,21 +1099,31 @@ def linear_spectrum_fitting(spectrum, params, elemental_lines=None,
         out, res = nnls_fit(y, matv)
 
     total_y = out * matv
-
-    # use ordered dict
-    result_dict = OrderedDict()
-
-    for i in range(len(total_list)):
-        if np.sum(total_y[:, i]) == 0:
-            continue
-        result_dict.update({total_list[i]: total_y[:, i]})
-    result_dict.update(background=bg)
-
     x = (params['e_offset']['value'] +
          params['e_linear']['value']*x +
          params['e_quadratic']['value'] * x**2)
 
-    return x, result_dict
+    if area_option:
+        result_dict = OrderedDict()
+        area_dict = OrderedDict()
+
+        for i in range(len(total_list)):
+            if np.sum(total_y[:, i]) == 0:
+                continue
+            result_dict.update({total_list[i]: total_y[:, i]})
+            area_dict.update({total_list[i]: out[i]*element_area[total_list[i]]})
+        result_dict.update(background=bg)
+        area_dict.update(background=np.sum(bg))
+        return x, result_dict, area_dict
+    else:
+        result_dict = OrderedDict()
+
+        for i in range(len(total_list)):
+            if np.sum(total_y[:, i]) == 0:
+                continue
+            result_dict.update({total_list[i]: total_y[:, i]})
+        result_dict.update(background=bg)
+        return x, result_dict
```

Comment:
This seems very strange to have optional return values. I would suggest using a different pattern